File: CourseCrafter_Selfmade/app/src/main/java/com/nfricke/coursecrafter_selfmade/ModulManager.java
package com.nfricke.coursecrafter_selfmade;

import java.util.ArrayList;

public class ModulManager extends ArrayList<Modul> {

    public int getIndexByName(String name) {
        for (int i = 0; i < this.size(); i++) {
            if (this.get(i).getModulName() == name) {
                return i;
            }
        }
        return -1;
    }
    public String[][] getByTagBlock(int tag, int block) {
        int size = 0;
        for (int i = 0; i < this.size(); i++) {
            for (int n = 0; n < this.get(i).getAnzahlVeranstaltungen(); n++) {
                if (this.get(i).getTag(n) == tag && this.get(i).getBlock(n) == block && this.get(i).isBelegt() == true) {
                    size = size + 1;
                }
            }
        }

        String[][] result = new String[size][2];
        int c = 0;
        for (int i = 0; i < this.size(); i++) {
            for (int n = 0; n < this.get(i).getAnzahlVeranstaltungen(); n++) {
                if (this.get(i).getTag(n) == tag && this.get(i).getBlock(n) == block && this.get(i).isBelegt() == true) {
                    result[c][0] = this.get(i).getModulName();
                    result[c][1] = this.get(i).getRaum(n);
                    c = c + 1;
                }
            }
        }
        return result;
    }

    public double durchschnitt(){
        ModulManager ergebnis = new ModulManager();
        for ( Modul s: this){
            if (s.getNote() != 0.0 && s.getNote() != 5.0){
                ergebnis.add(s);
            }
        }
        double durchschnit = 0;
        for ( Modul s: ergebnis){
            durchschnit = durchschnit + s.getNote();
        }
        durchschnit = durchschnit / ergebnis.size();

        return Math.round(durchschnit * 100.0) / 100.0;
    }

    public boolean nameIstVorhanden(String name){
        for ( Modul s: this){
            if (s.getModulName().equals(name))
                return true;
        }
        return false;
    }
    /*
    public void printTest() {
        for (Modul s : this) {
            s.printTest();
        }
    }
     */
}

File: CourseCrafter/app/src/main/java/com/nfricke/coursecrafter/ModulManager.kt
package com.nfricke.testapp03


class ModulManager : ArrayList<Modul>() {

    fun getByTagBlock(tag: Int, block: Int): Array<String> {
        val result = emptyArray<String>()
        var c = 0
        for (i in this.indices) {
            for (n in 0 until this[i].getAnzahlVeranstaltungen()) {
                if (this[i].getTag(n) == tag && (this[i].getBlock(n) == block)) {
                    result[c] = this[i].getModulName()
                    c++
                }
            }
        }
        return result
    }

    fun nameIstVorhanden(name: String?): Boolean {
        for (s in this) {
            if (s.getModulName() == name) return true
        }
        return false
    }

    fun getIndexByName(name: String): Int {
        for (i in this.indices) {
            if (this[i].getModulName() === name) {
                return i
            }
        }
        return -1
    }


    fun printTest() {
        for (s in this) {
            s.printTest()
        }
    }
}